cmd/yggdrasil/main.go

```diff
@@ -244,22 +248,29 @@ func main() {
 	// Setup the Yggdrasil node itself. The node{} type includes a Core, so we
 	// don't need to create this manually.
 	n := node{}
-	// Now that we have a working configuration, we can now actually start
-	// Yggdrasil. This will start the router, switch, DHT node, TCP and UDP
-	// sockets, TUN/TAP adapter and multicast discovery port.
-	if err := n.core.Start(cfg, logger); err != nil {
+	// Before we start the node, set the TUN/TAP to be our router adapter
+	n.core.SetRouterAdapter(&n.tuntap)
```

Comment:
I still feel like this is backwards. The core/router should have in/out channels or functions (functions is probably better, since you can't accidentally overwrite them), and then have the tun/tap make use of those, instead of injecting a dependency after the fact. But that requires a little care to make sure that whatever the `Read()` equivalent is doesn't cause the core to either block or drop traffic unnecessarily if it's not being called often enough. So I'm not sure we should try to deal with it now. Just a reminder to take another look at this later, since it's ultimately how the user is going to interact and send/receive traffic.